internal/service/vmservice/utils.go

```diff
@@ -78,8 +117,14 @@ func shouldUpdateNetworkDevices(machineScope *scope.MachineScope) bool {
 		if net0 == "" {
 			return true
 		}
-		model, bridge := extractNetworkModelAndBridge(net0)
-		if model != *machineScope.ProxmoxMachine.Spec.Network.Default.Model || bridge != machineScope.ProxmoxMachine.Spec.Network.Default.Bridge {
+
+		desiredDefault := *machineScope.ProxmoxMachine.Spec.Network.Default
+
+		model := extractNetworkModel(net0)
+		bridge := extractNetworkBridge(net0)
+		mtu := extractNetworkMTU(net0)
+
+		if model != *desiredDefault.Model || bridge != desiredDefault.Bridge || mtu != *desiredDefault.MTU {
```

Comment:
Do we need to check if mtu was specified first?

Reply:
I dont think so because it is said to `default=1500` in the CRD (similar how the Model is set virtio as default)

Reply:
https://github.com/ionos-cloud/cluster-api-provider-proxmox/pull/42#discussion_r1431642877